src/composables/multisigAccounts.ts

```diff
@@ -0,0 +1,294 @@
+import { computed, ref } from '@vue/composition-api';
+import type { Store } from 'vuex';
+import multisigContract from '@aeternity/ga-multisig-contract/SimpleGAMultiSig.aes';
+import { uniqBy } from 'lodash-es';
+import camelcaseKeysDeep from 'camelcase-keys-deep';
+import {
+  MemoryAccount, Crypto,
+} from '@aeternity/aepp-sdk';
+import { decode } from '@aeternity/aepp-sdk/es/tx/builder/helpers';
+
+import {
+  fetchJson,
+  postJson,
+  LOCAL_STORAGE_PREFIX,
+  MULTISIG_CREATION_STEPS,
+} from '../popup/utils';
+import { createPollingBasedOnMountedComponents } from './composablesHelpers';
+import type {
+  IAccount,
+  INetwork,
+  IMultisigCreationStep,
+  IMultisigAccountBase,
+  IMultisigAccount,
+} from '../types';
+import { useSdk } from './sdk';
+
+interface UseMultisigOptions {
+  /**
+   * TODO: Temporary solution to avoid dependency circle
+   */
+  store: Store<any>
+}
+
+const POLLING_INTERVAL = 5000;
+const MULTISIG_TRANSACTION_EXPIRATION_HEIGHT = 50;
+
+const LOCAL_STORAGE_MULTISIG_KEY = `${LOCAL_STORAGE_PREFIX}_multisig`;
+
+function storeMultisigAccounts(multisigAccounts: IMultisigAccount[], networkId: string) {
+  window.localStorage
+    .setItem(`${LOCAL_STORAGE_MULTISIG_KEY}_${networkId}`, JSON.stringify(multisigAccounts));
+}
+
+function getStoredMultisigAccounts(networkId: string): IMultisigAccount[] {
+  const storedMultisig = window.localStorage.getItem(`${LOCAL_STORAGE_MULTISIG_KEY}_${networkId}`);
+  return storedMultisig ? JSON.parse(storedMultisig) : [];
+}
+
+const multisigAccounts = ref(<IMultisigAccount[]>[]);
+const activeMultisigAccountId = ref('');
+const isMultisigDashboard = ref(false);
+
+const initPollingWatcher = createPollingBasedOnMountedComponents();
+
+export function useMultisigAccounts({ store }: UseMultisigOptions) {
+  const { getDrySdk, getSdk } = useSdk({ store });
+
+  const accounts = computed<IAccount[]>(() => store.getters.accounts);
+  const activeNetwork = computed<INetwork>(() => store.getters.activeNetwork);
+
+  multisigAccounts.value = getStoredMultisigAccounts(activeNetwork.value.networkId);
```

Comment:
I'm afraid of performance impact of this action. The `multisigAccounts` composable could be used within many components at the same time. So every time the component fires `useMultisigAccounts` it will try to get the data from local storage and parse it.

I suggest to replace this with logic that would check if the data for the currently selected network is empty and only then retrieve it from the local storage.

Or modify the `getStoredMultisigAccounts` to cache the data already retrieved for the selected network.

What do you think?